search_service/proxy/atlas.py

```diff
@@ -45,71 +35,187 @@ def __init__(self, *,
         self.page_size = page_size
 
     @staticmethod
-    def _entities(collections: EntityCollection) -> List[Entity]:
+    def _extract_entities(collections: EntityCollection) -> List[Entity]:
         """
         Helper method for flattening all collections from {collections}
+
         :return: list of all entities
         """
         entities: List[Entity] = []
+
         for collection in collections:
             entities.extend(collection.entities)
         return entities
 
-    def _parse_results(self, response: EntityCollection) -> List[Table]:
+    def _prepare_tables(self, response: EntityCollection, enhance_metadata: bool = False) -> List[Table]:
         """
-        based on an atlas {response} with table entities, we map the required information
-        :return: list of tables
+        Based on an Atlas {response} with table entities, we render Table objects.
+
+        :param response: Collection of Atlas Entities
+        :param enhance_metadata: Should Atlas be queried to acquire complete entity definitions (search might not
```

Comment:
Can you clarify if this means that when `enhance_metadata=True` then search might not return all available attributes or is it the other way around?

Reply:
when using advanced search very minimal set of attributes is returned back. so enhance_metadata set to True takes guids of all returned tables (entities) and queries Atlas for complete information (along with relationship attributes and so on).